File: CS/CustomCommand/CustomCommands.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraRichEdit.Commands;
using System.Windows.Forms;
using DevExpress.XtraRichEdit.Services;
using DevExpress.XtraRichEdit;
using DevExpress.Utils;

namespace CustomCommand
{
    #region customsavecommand
    public class CustomSaveDocumentCommand : SaveDocumentCommand
    {
        public CustomSaveDocumentCommand(IRichEditControl control)
            : base(control)
        {
        }

        protected override void ExecuteCore()
        {
            if (Control.Document.Paragraphs.Count > 7)
                base.ExecuteCore();
            else MessageBox.Show(@"You should type at least 7 paragraphs
  to be able to save the document.",
                "Please be creative", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }
    }
    #endregion customsavecommand


    #region #iricheditcommandfactoryservice
    public class CustomRichEditCommandFactoryService : IRichEditCommandFactoryService
    {
        readonly IRichEditCommandFactoryService service;
        readonly RichEditControl control;

        public CustomRichEditCommandFactoryService(RichEditControl control, 
            IRichEditCommandFactoryService service)
        {
            Guard.ArgumentNotNull(control, "control");
            Guard.ArgumentNotNull(service, "service");
            this.control = control;
            this.service = service;
        }

        #region IRichEditCommandFactoryService Members
        public RichEditCommand CreateCommand(RichEditCommandId id)
        {
            if (id == RichEditCommandId.FileSave)
                return new CustomSaveDocumentCommand(control);

            return service.CreateCommand(id);
        }
        #endregion
    }
    #endregion #iricheditcommandfactoryservice


}

File: VB/CustomCommand/CustomCommands.vb
Imports DevExpress.XtraRichEdit.Commands
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit.Services
Imports DevExpress.XtraRichEdit
Imports DevExpress.Utils

Namespace CustomCommand

#Region "customsavecommand"
    Public Class CustomSaveDocumentCommand
        Inherits SaveDocumentCommand

        Public Sub New(ByVal control As IRichEditControl)
            MyBase.New(control)
        End Sub

        Protected Overrides Sub ExecuteCore()
            If Control.Document.Paragraphs.Count > 7 Then
                MyBase.ExecuteCore()
            Else
                MessageBox.Show("You should type at least 7 paragraphs
  to be able to save the document.", "Please be creative", MessageBoxButtons.OK, MessageBoxIcon.Information)
            End If
        End Sub
    End Class

#End Region  ' customsavecommand
#Region "#iricheditcommandfactoryservice"
    Public Class CustomRichEditCommandFactoryService
        Implements IRichEditCommandFactoryService

        Private ReadOnly service As IRichEditCommandFactoryService

        Private ReadOnly control As RichEditControl

        Public Sub New(ByVal control As RichEditControl, ByVal service As IRichEditCommandFactoryService)
            Guard.ArgumentNotNull(control, "control")
            Guard.ArgumentNotNull(service, "service")
            Me.control = control
            Me.service = service
        End Sub

#Region "IRichEditCommandFactoryService Members"
        Public Function CreateCommand(ByVal id As RichEditCommandId) As RichEditCommand Implements IRichEditCommandFactoryService.CreateCommand
            If id = RichEditCommandId.FileSave Then Return New CustomSaveDocumentCommand(control)
            Return service.CreateCommand(id)
        End Function
#End Region
    End Class
#End Region  ' #iricheditcommandfactoryservice
End Namespace
